index.rst

```diff
@@ -1,61 +1,263 @@
+
+`Kafka`_ is an open source event streaming platform that is broadly used in industry.
+In fact, it is already an integral part of the Rubin-OCS, as it is used in the EFD to transport the data from DDS to influxDB (:sqr:`034` :cite:`SQR-034`).
+
+.. _Kafka: https://kafka.apache.org
+
+The fact that we are already using Kafka in the system reliably to ingest data into the EFD gives us confidence that it is, at the very least, able to handle the overall data throughput.
+Our main concern is than to verify that Kafka can handle the latency requirements of our system.
+In principle, Kafka is advertised as a "real-time" system and numerous benchmarks exists online showing it can reach latencies at the millisecond regime.
+Nevertheless, it is unclear those benchmarks would be applied to our systems constrains, giving the typical message size, network architecture and other relevant factors.
+
+We then proceeded to perform benchmarks with the intention to evaluate Kafka's performance considering our system architecture.
+The results, which are detailed in :tstn:`033`, are encouraging.
+In summary, we obtain similar latency levels for both Kafka and DDS.
+In terms of throughput, DDS is considerably better than Kafka for smaller messages, though we obtain similar values for larger messages.
+
+Overall, our detailed study shows that Kafka would be a viable option for replacing DDS as the middleware technology in our system.
+For the full technical report see :tstn:`033`.
 
-.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
+Summary
+=======
 
-.. sectnum::
+After considerable effort fine tunning the DDS middleware configuration, we were finally able to obtain a stable system, that is capable of operating at large scale with low middleware-related failure rate.
+At the current advanced state of the project, which is approaching its final construction stages, one might be tempted to accept this part of the project as concluded.
 
-.. TODO: Delete the note below before merging new content to the main branch.
+Nevertheless, as we demonstrated, there are a number of issues hiding underneath that may pose significant problems in the future, or even be seen as violating system requirements.
 
-.. note::
+Overall our experience with DDS has been, to say the least, underwhelming.
+Even thought the technology is capable of achieving impressive throughput and latency, in reality, it proved to be extremely cumbersome and hard to manage and debug on large scale systems.
+On top of if all we also face a potentially end-of-life cycle of the adopted library, which makes the problem considerably worse.
 
-   **This technote is not yet published.**
+After extensive exploring different alternatives to the problem we propose a potential long term solution to the problem, namely, replace DDS by the already in-use Kafka.
```

Comment:
"After extensive exploration of" but I worry that you will get push back on this. We did not do extensive research. We did some research -- enough to convince ourselves. The question is whether it will be enough to convince others. In any case I strongly recommend softening this, perhaps to something along the lines:

"After exploring different solutions to the problem of long-term maintenance of our middleware, we propose to replace DDS by the already in-use Kafka".

Reply:
Yeah, I would agree with Russell on this.